conjure-java-core/src/main/java/com/palantir/conjure/java/services/UndertowServiceInterfaceGenerator.java

```diff
@@ -96,17 +99,69 @@ private MethodSpec generateServiceInterfaceMethod(
 
         endpointDef.getDeprecated().ifPresent(deprecatedDocsValue -> methodBuilder.addAnnotation(Deprecated.class));
 
-        methodBuilder.addJavadoc("$L", ServiceGenerators.getJavaDocWithRequestLine(endpointDef));
+        addJavaDoc(methodBuilder, endpointDef, options.packagePrefix());
 
         if (UndertowTypeFunctions.isAsync(endpointDef, options)) {
             methodBuilder.returns(UndertowTypeFunctions.getAsyncReturnType(endpointDef, returnTypeMapper, options));
         } else {
             endpointDef.getReturns().ifPresent(type -> methodBuilder.returns(returnTypeMapper.getClassName(type)));
         }
 
+        methodBuilder.addExceptions(endpointDef.getErrors().stream()
+                .map(endpointError -> {
+                    ErrorTypeName errorTypeName = endpointError.getError();
+                    return ClassName.get(
+                            Packages.getPrefixedPackage(errorTypeName.getPackage(), options.packagePrefix()),
+                            "Server" + errorTypeName.getNamespace() + "Errors",
+                            errorTypeName.getName());
+                })
+                .collect(Collectors.toList()));
+
         return methodBuilder.build();
     }
 
+    private static void addJavaDoc(
+            MethodSpec.Builder methodBuilder,
+            EndpointDefinition endpointDefinition,
+            Optional<String> maybePackagePrefix) {
+        Optional<String> depr = endpointDefinition.getDeprecated().map(Javadoc::getDeprecatedJavadoc);
+        Optional<String> incDoc = Javadoc.getIncubatingJavadoc(endpointDefinition.getTags());
+        Optional<String> docs = endpointDefinition.getDocs().map(Javadoc::render);
+        Optional<String> requestLine = Optional.of(
+                Javadoc.getRequestLine(endpointDefinition.getHttpMethod(), endpointDefinition.getHttpPath()));
+        Optional<String> params = Optional.ofNullable(Strings.emptyToNull(endpointDefinition.getArgs().stream()
+                .flatMap(argument -> Javadoc.getParameterJavadoc(argument, endpointDefinition).stream())
+                .collect(Collectors.joining("\n"))));
+        StringBuilder sb = new StringBuilder();
```

Comment:
Instead of collecting to this stringbuilder, would it make sense to individually apply each component to `methodBuilder.addJavadoc` if non-empty? It's possible that might do odd things to spacing, but worth trying, I think it could simplify this quite a bit